unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,205 @@
+
+      REQUIRE(symbol_type.id()==ID_struct);
+      class_typet class_type=to_class_type(symbol_type);
+      REQUIRE(class_type.is_class());
+
+      REQUIRE(class_type.has_component("key"));
+      const auto &first_component=class_type.get_component("key");
+      REQUIRE(is_java_generic_inst_parameter(first_component.type()));
+      REQUIRE(class_type.has_component("value"));
+      const auto &second_component=class_type.get_component("value");
+      REQUIRE(is_java_generic_inst_parameter(second_component.type()));
+    }
+  }
+
+}
+
+
+SCENARIO(
+  "generate_java_generic_type_from_file_uninstantiated_param",
+  "[CORE][java_bytecode][generate_java_generic_type]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+  GIVEN("A generic java type with a field that refers to another generic with"
+          " an uninstantiated parameter.")
+  {
+    java_lang->parse(java_code_stream, "generic_unknown_field.class");
+
+    symbol_tablet new_symbol_table;
+    java_lang->typecheck(new_symbol_table, "");
+
+    java_lang->final(new_symbol_table);
+
+    REQUIRE_FALSE(new_symbol_table.has_symbol
+      ("java::generic_unknown_field$element<T>"));
+  }
+}
+
+
+SCENARIO(
+  "generate_java_generic_type_from_file_two_instances",
+  "[CORE][java_bytecode][generate_java_generic_type]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+  GIVEN("A generic java type with a field that refers to another generic with"
+          " an uninstantiated parameter.")
+  {
+    java_lang->parse(java_code_stream, "generic_two_instances.class");
+
+    symbol_tablet new_symbol_table;
+    java_lang->typecheck(new_symbol_table, "");
+
+    java_lang->final(new_symbol_table);
+
+    REQUIRE(new_symbol_table.has_symbol
+      ("java::generic_two_instances$element<java::java.lang.Boolean>"));
+    REQUIRE(new_symbol_table.has_symbol
+      ("java::generic_two_instances$element<java::java.lang.Integer>"));
```

Comment:
Please tighten as with the above tests to check that the components of the created class are of the correct type